doc/SAI-Proposal-Prefix-Compression.md

```diff
@@ -0,0 +1,271 @@
+    sai_get_prefix_compression_table_attribute_fn       get_prefix_compression_table_attribute;
+    sai_create_prefix_compression_entry_fn              create_prefix_compression_entry;
+    sai_remove_prefix_compression_entry_fn              remove_prefix_compression_entry;
+    sai_set_prefix_compression_entry_attribute_fn       set_prefix_compression_entry_attribute;
+    sai_get_prefix_compression_entry_attribute_fn       get_prefix_compression_entry_attribute;
+    sai_bulk_create_prefix_compression_entry_fn         create_prefix_compression_entries;
+    sai_bulk_remove_prefix_compression_entry_fn         remove_prefix_compression_entries;
+} sai_prefix_compression_api_t;
+```
+
+## 3 Examples
+
+## 3.0.1 Create Prefix Compression Table with entries
+
+```c
+sai_attr_table_list[];
+sai_attr_table_list[0].id = SAI_PREFIX_COMPRESSION_TABLE_ATTR_DEFAULT_ENTRY_BINCODE;
+sai_attr_table_list[0].value.u32 = 9000;
+attr_table_count = 0;
+sai_create_prefix_compression_table_fn(
+    &src_prefix_compression_table_id,
+    switch_id,
+    attr_table_count,
+    sai_attr_table_list);
+
+// Example: Create Prefix Compression Entries
+// IPV4 First Entry
+sai_prefix_compression_entry_t entry_v4_1;
+entry_v4_1.switch_id = switch_id;
+entry_v4_1.prefix_table_id = src_prefix_compression_table_id;
+entry_v4_1.prefix.addr_family = SAI_IP_ADDR_FAMILY_IPV4;
+entry_v4_1.prefix.addr.ipv4 = "1.1.1.1";
+entry_v4_1.prefix_mask.ipv4 = "255.255.255.0";
+sai_entry_list[0].id = SAI_PREFIX_COMPRESSION_ENTRY_ATTR_BINCODE;
+sai_entry_list[0].value.u32 = 2;
+
+// IPV4 Second Entry
+sai_prefix_compression_entry_t entry_v4_1;
+entry_v4_1.switch_id = switch_id;
+entry_v4_1.prefix_table_id = src_prefix_compression_table_id;
+entry_v4_1.prefix.addr_family = SAI_IP_ADDR_FAMILY_IPV4;
+entry_v4_1.prefix.addr.ipv4 = "2.2.2.1";
+entry_v4_1.prefix_mask.ipv4 = "255.255.255.0";
+sai_entries_attribute_list[0].id = SAI_PREFIX_COMPRESSION_ENTRY_ATTR_BINCODE;
+sai_entries_attribute_list[0].value.u32 = 800;
+
+// IPV6 Entry
+sai_prefix_compression_entry_t entry_v6_1;
+entry_v6_1.switch_id = switch_id;
+entry_v6_1.prefix_table_id = src_prefix_compression_table_id;
+entry_v6_1.prefix.addr_family = SAI_IP_ADDR_FAMILY_IPV6;
+entry_v6_1.prefix.addr.ipv4 ="2001:1::4";
+entry_v6_1.prefix_mask = "ffff:ffff:ffff:ffff:ffff:ffff:ffff:fffc";
+sai_entries_attribute_list[1].id = SAI_PREFIX_COMPRESSION_ENTRY_ATTR_BINCODE;
+sai_entries_attribute_list[1].value.u32 = 4;
+
+entries_list = [entry_v4_2, entry_v6_1];
+entries_list_size = 2;
+sai_status_t bulk_status;
+
+// Adding a single entry to a table
+sai_attr_list_count = 1;
+sai_create_prefix_compression_entry_fn(
+    entry_v4_1,
+    sai_attr_list,
+    sai_attr_list_count);
+
+// Adding using bulk add
+attr_entries_count = 2;
+sai_bulk_create_prefix_compression_entry_fn(
+    entries_list_size,
+    entries_list,
+    attr_entries_count,
+    sai_entries_attribute_list,
+    SAI_BULK_OP_ERROR_MODE_STOP_ON_ERROR,
+    bulk_status);
+
```

Comment:
Please consider including the ACL table and ACL entry usage of the prefix table and prefix entry respectively.

Reply:
I created a separate document in ACL on ACL using this feature. However, this feature standalone does not depend on ACL. For that reason we should not be listing an example of it in this part of the documenations.

I want prefix compression to be though off as a separate feature than ACL. This is a standalone feature that other features can use. In this case I am using it with ACL, in the future there might be another use of it with another feature.

please refer to [doc/ACL/SAI-Proposal-ACL-Bincode.md]

Reply:
@rck-innovium, I updated the documentation. I combined both in a single document. Please let me know if this meets your request and resolve if it does